src/rgw/rgw_rest_realm.cc

```diff
@@ -84,9 +84,9 @@ void RGWOp_Period_Post::execute()
   period.init(cct, store, false);
 
   // decode the period from input
-#define PERIOD_MAX_LEN 4096
+  const auto max_size = cct->_conf->rgw_max_put_param_size;
```

Comment:
yeah, lets just change this to `const uint64_t max_size = 1 * 1024 * 1024;` instead of backporting `rgw_max_put_param_size`